app/frontend/tokenRegistry/tokenRegistryApi.ts

```diff
@@ -0,0 +1,46 @@
+import request from '../wallet/helpers/request'
+import {ADALITE_CONFIG} from '../config'
+import {RegisteredTokenMetadata} from '../types'
+import {TokenRegistryApi} from './types'
+
+export default (): TokenRegistryApi => {
+  const fetchTokensMetadata = (subjects: string[]): Promise<any> => {
+    const url = `${ADALITE_CONFIG.ADALITE_SERVER_URL}/api/tokenRegistry/getTokensMetadata`
+    const requestBody = {subjects}
+    return request(url, 'POST', JSON.stringify(requestBody), {
+      'Content-Type': 'application/json',
+    })
+  }
+
+  const parseTokensMetadata = (toParse: any): {[subject: string]: RegisteredTokenMetadata} => {
+    if (toParse?.Right) {
+      return toParse.Right.reduce((acc, tokenMetadata) => {
+        const {subject, description, ticker, url, logoHex} = tokenMetadata
+        acc[subject] = {subject, description, ticker, url, logoHex}
+        acc[tokenMetadata.subject] = {
+          subject: tokenMetadata.subject,
+          description: tokenMetadata.description.value,
+          ticker: tokenMetadata?.ticker?.value,
+          url: tokenMetadata?.url?.value,
+          logoHex: tokenMetadata?.logo?.value,
+          decimals: tokenMetadata?.decimals?.value,
+        }
+        return acc
+      }, {})
+    } else {
+      return {}
+    }
+  }
+
+  const getTokensMetadata = async (
+    subjects: string[]
```

Comment:
Given this function faces towards our wallet logic, I'd consider not exposing the raw "subject" in the function signature, but rather let it take our Token type and do the transformation to "subject" within the module, to not leak this internal representation of tokens towards higher level logic - the benefit is that the function signature would be less cryptic (no generic strings/custom one-off types) and the transformation logic would be contained/centralized within this module

Reply:
Refactored